src/lib/markdowndb.ts

```diff
@@ -80,14 +82,36 @@ export class MarkdownDB {
     pathToUrlResolver?: (filePath: string) => string;
     customConfig?: CustomConfig;
   }) {
-    await resetDatabaseTables(this.db);
-
     const fileObjects = indexFolder(
       folderPath,
       pathToUrlResolver,
       customConfig,
       ignorePatterns
     );
+    await this.saveDataToDisk(fileObjects);
+  }
+
+  private async saveDataToDisk(fileObjects: FileInfo[]) {
+    await resetDatabaseTables(this.db);
+    const properties = getUniqueProperties(fileObjects);
+    MddbFile.deleteTable(this.db);
+    const defaultProperties = [
+      "_id",
+      "file_path",
+      "extension",
+      "url_path",
+      "filetype",
+      "metadata",
+      "tags",
+      "links",
+    ];
+    await MddbFile.createTable(
+      this.db,
+      properties.filter(
+        (property) => defaultProperties.indexOf(property) === -1
+      )
```

Comment:
It doesn't look good, move any detailed logic related to table creation the createTable method.